tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/Analyzer.scala

```diff
@@ -349,7 +350,80 @@ final class Analyzer(semantics: Semantics, outputMode: OutputMode,
           None
         }
       }
-      loop(this)
+      val existing =
+        if (isScalaClass) loop(this)
+        else methodInfos.get(methodName).filter(!_.isAbstract)
```

Comment:
So if I read this correctly, static calls on classes have inheritance while on interfaces they do not. Why is this so? Is this a parallel to JVM bytecode?